catroid/src/org/catrobat/catroid/content/Project.java

```diff
@@ -424,4 +421,33 @@ public boolean containsSprite(Sprite selectedSprite) {
 		}
 		return false;
 	}
+
+	public void addSprite(int spriteIndex, Sprite sprite) {
+		spriteList.add(spriteIndex, sprite);
+	}
+
+	public void refreshSpriteReferences() {
+		Project project = ProjectManager.getInstance().getCurrentProject();
+		for (Brick brick : project.getAllBricks()) {
+			if (!(brick instanceof PointToBrick)) {
+				continue;
+			}
+			PointToBrick pointToBrick = (PointToBrick) brick;
+			Sprite pointedSprite = pointToBrick.getPointedObject();
+			if (pointedSprite == null) {
+				continue;
+			}
+
+			Sprite newSprite = getSpriteBySpriteName(pointToBrick.getPointedObject());
```

Comment:
The PointToBrick can only point to other objects, not to a group at all (the group is just a layer around the Sprites).
And the method refreshSpriteReferences(..) is needed because all PointTo-Bricks have a reference to the old (ready for GC) sprites - that need to be set to the newly created Sprites ;)